### ISSUE TITLE
Definition of a package

### ISSUE THREAD

**Issue by e9gille** [theaplroom/APM#1]
A package consists of:

- a single, top level class or namespace
- a configuration file that describes the package
- [optional] extra resources (binary files, stylesheets, executables, etc.) 

A package may depend on other packages, in which case these dependencies should be listed in the configuration file. The dependencies (other packages) should not be included in the package but will be retrieved by the APM.

**Comment by PaulMansour** [comment]
As you have noted, a a package looks suspiciously like a an Acre Desktop project, as one would expect. 

What are the differences? I think:

1.  Rather than a source folder of code in files (the /APLSource folder) we have a source dws file APLSource.dws, for example.

1. Much of the source code may actually be removed in the package - tests, stuff that supports development, etc.

1. Many of the extra resources may be removed as well, as they may support tests and other things not needed for distribution.

1. There is no .acre or .git folder, etc.

Other than this, I think they are or can be almost exactly the same.  In fact, I think Acre-Desktop is going to need to be able to load a package as if it were a project.

Consider me working on FlipDB and Rumba, two of my own projects I may work on concurrently. I use Kai's MarkAPL, which I do not work on.  When I open FlipDB for development I want Acre to open all the dependent projects, but one of them is on my disk as a package, not a project.  I think Acre should not care, it should be able to load a package just as easily as opening a project. 

Of course a package is always "loaded" even if you ask Acre to "open" it.  

A project  must be able load a package, but I don't think there is any need for a package to load a project.

**Comment by e9gille** [comment]
Agreed, a package would not depend on a project, it would only ever list its dependencies as packages.

I'm starting to realize that the overlap of Acre and APM is too big. I'm not sure what the point of separating them would be anymore. Put it this way:
1. Is there any reason anyone would opt out of Acre, but use APM to manage dependencies?
2. If so, and if Acre is the APM (as in all required features are built into it) what would that entail?

Answering my own questions:
Maybe if you have a workspace with lots of things in the root, configuring Acre to use # as the project space you can no longer bring in other projects. If you don't go through the trouble of refactoring your code into modular namespaces it means you can't use packages either.

So, if an Acre project could be configured in a way that it doesn't have a project space that might work, but I don't even want to think about the implications of that.
